pkg/oc/cli/admin/ipfailover/ipfailover.go

```diff
@@ -60,189 +57,245 @@ var (
 	// configurations can run on the same node a different ServicePort is used
 	// for each configuration.
 	// In the future, this may be changed to pod anti-affinity.
-	options.ServicePort = options.ServicePort + options.VRRPIDOffset
+	o.ConfigOptions.ServicePort = o.ConfigOptions.ServicePort + o.ConfigOptions.VRRPIDOffset
 
-	restMapper, err := f.ToRESTMapper()
+	o.RESTMapper, err = f.ToRESTMapper()
 	if err != nil {
 		return err
 	}
 	clientConfig, err := f.ToRESTConfig()
 	if err != nil {
 		return err
 	}
-	dynamicClient, err := dynamic.NewForConfig(clientConfig)
+	o.DynamicClient, err = dynamic.NewForConfig(clientConfig)
 	if err != nil {
 		return err
 	}
 
-	options.Action.Bulk.Scheme = legacyscheme.Scheme
-	options.Action.Bulk.Op = configcmd.Creator{
-		Client:     dynamicClient,
-		RESTMapper: restMapper,
-	}.Create
-
-	if err := ipfailover.ValidateCmdOptions(options); err != nil {
-		return err
+	o.ConfigOptions.ConfiguratorPlugin, err = keepalived.NewIPFailoverConfiguratorPlugin(name, f, o.ConfigOptions)
+	if err != nil {
+		return fmt.Errorf("IPFailoverConfigurator %q plugin error: %v", o.ConfigOptions.Type, err)
 	}
 
-	p, err := getPlugin(name, f, options)
+	o.SecurityClient, err = securityv1typedclient.NewForConfig(clientConfig)
 	if err != nil {
 		return err
 	}
 
-	list, err := p.Generate()
-	if err != nil {
+	return nil
+}
+
+func (o *IPFailoverConfigCmdOptions) Validate() error {
+	if err := validateVirtualIPs(o.ConfigOptions.VirtualIPs); err != nil {
 		return err
 	}
 
-	namespace, _, err := f.ToRawKubeConfigLoader().Namespace()
-	if err != nil {
-		return err
+	if o.ConfigOptions.Type != ipfailover.DefaultType {
+		return fmt.Errorf("no plugins available to handle type %q", o.ConfigOptions.Type)
+	}
+
+	if err := validateServiceAccount(o.SecurityClient, o.ConfigOptions.ServiceAccount); err != nil {
+		return fmt.Errorf("ipfailover could not be created, invalid service account name: %v", err)
+	}
+
+	return nil
+}
+
+//  Get configuration name - argv[1].
+func configurationName(args []string) (string, error) {
+	name := ipfailover.DefaultName
+
+	switch len(args) {
+	case 0:
+		// Do nothing - use default name.
+	case 1:
+		name = args[0]
+	default:
+		return "", fmt.Errorf("zero arguments or an argument containing a configuration name is required")
 	}
-	securityClient, err := securityclientinternal.NewForConfig(clientConfig)
+
+	return name, nil
+}
+
+// Run runs the ipfailover command.
+func (o *IPFailoverConfigCmdOptions) Run() error {
+	list, err := o.ConfigOptions.ConfiguratorPlugin.Generate()
 	if err != nil {
 		return err
 	}
-	if err := validateServiceAccount(securityClient, namespace, options.ServiceAccount); err != nil {
-		return fmt.Errorf("ipfailover could not be created; %v", err)
-	}
 
-	configList := []runtime.Object{
-		&kapi.ServiceAccount{ObjectMeta: metav1.ObjectMeta{Name: options.ServiceAccount}},
+	configList := []runtime.RawExtension{
```

Comment:
Instead of dealing with `corev1.List`, wouldn't it be simpler to work with `[]runtime.Object`?

Reply:
Will do